_posts/en/newsletters/2018-10-09-newsletter.md

```diff
@@ -0,0 +1,283 @@
+time-warp attack to allow both upgraded nodes and old nodes to gain the
+same capacity increase and see all the same transactions for validation
+while simultaneously slightly reducing mining centralization risk.
+Upgraded nodes would validate one or more extension blocks providing
+additional block space with a 15 minute average interval, but they'd
+also restrict the time stamps in legacy blocks to ensure a permanent
+(but limited) time warp attack increased the frequency of legacy blocks
+enough to allow them to include the same transactions that previously
+appeared in the extension blocks.
+
+*[video][vid forward blocks], [transcript][tx forward blocks],
+[paper][paper forward blocks], soft fork*
+
+### Compact multi-signatures for smaller blockchains
+
+*Research by Dan Boneh, Manu Drijvers and Gregory Neven*
+
+This talk describes an alternative to the Bellare-Neven-based Schnorr
+signature scheme described in the [MuSig paper][] and [proposed as a
+standard][schnorr pre-bip] for Bitcoin by Pieter Wuille.  The
+alternative method makes use of [pairing-based cryptography][],
+specifically an adaptation of the [Boneh-Lynn-Shacham (BLS) signature
+scheme][bls sigs].  Although pairing-based schemes require an additional
+fundamental security assumption beyond those made by both Bitcoin's
+current ECDSA scheme and proposed Schnorr scheme, the authors present
+evidence that their scheme would produce smaller signatures in general,
+be much more computationally efficient in some multisignature use cases,
+and make it possible to prove which members of the set of signers
+actually worked together to create a threshold signature (i.e.  k-of-m
+signers, e.g. 2-of-3 multisig).
+
+*[video][vid bls msig], [transcript][tx bls msig], [paper
+(pre-print)][paper bls msig], soft fork using segwit script upgrade
+mechanism*
+
+### Accumulators: a scalable drop-in replacement for merkle trees
+
+*Research by Benedikt Bünz, Benjamin Fisch and Dan Boneh*
+
+In Bitcoin and other cryptocurrencies, scalable commitments to sets of
+information---such as transactions or UTXOs---are normally made using
+merkle trees that allow proving an element is a member of the set by
+generating a proof whose size and validation cost is roughly *log2(n)*
+for a set of *n* elements.
+
+This talk describes an alternative method based on RSA accumulators that
+provides many potential benefits: the size of a proof is constant no
+matter how many elements are members of the set, adding or removing
+elements from an accumulator can be efficiently batched (e.g. one update
+per block), and validation that one or more elements are members of a
+set could potentially be much faster (although this has not yet been
+fully tested).
+
+*[video][vid accumulators], [transcript][tx accumulators], soft fork for
+new applications such as UTXO set commitments*
+
+### Scriptless ECDSA
+
+*Research by Conner Fromknecht*
+
+At Scaling Bitcoin IV (Stanford 2017), Andrew Poelstra
```

Comment:
This sentence is a bit long and unwieldy. Consider splitting to something like:

> At Scaling Bitcoin IV (Stanford 2017), Andrew Poelstra [presented][scriptless scripts transcript] his existing "scriptless scripts" work. *Scriptless scripts* allow outputs to be encumbered with conditions in the same way that is currently possible with Bitcoin Script, but by using only Schnorr-based keys and signatures.

Reply:
Actually, I think this introductory paragraph should be less focused on general scriptless scripts, and more on reblinding the challenge in lightning routes. As far as I'm aware, this concept was introduced in Andrew Poesltra's mailing list post here: https://lists.launchpad.net/mimblewimble/msg00086.html and required the linear property of Schnorr signatures to be able to aggregate 2-of-2 multisignatures into a single signatures.

I don't think that there are efficient ways of doing general threshold signatures in ECDSA, but last year Yehuda Lindell published a paper for efficient 2-of-2 signatures (https://eprint.iacr.org/2017/552), which was applied to lightning routes in the Multi-hop locks paper (https://eprint.iacr.org/2018/472.pdf). This talk was about making a working implementation of that scheme.

The non-linearity of ECDSA makes those 2-of-2 aggregate signatures a lot more complex to produce, but Conner argued that:
- we'll want the reblinded challenge in lightning when/if schnorr signatures are available
- doing it now with ECDSA allows a graceful upgrade path (since 2P-ECDSA lightning channels could interop with Schnorr lightning channels)
- it may be better to just do it now with ECDSA while the lightning network is small.

Reply:
I've just confirmed the above with Andrew Poelstra.